src/modules/browser/speech-synthesis/BrowserSpeechSettings.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+
+import { Option, FormControl, Select, Switch, Typography, Box, IconButton } from '@mui/joy';
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
+import CloseRounded from '@mui/icons-material/CloseRounded';
+import { addSnackbar } from '~/common/components/snackbar/useSnackbarsStore';
+
+import { FormLabelStart } from '~/common/components/forms/FormLabelStart';
+import { useBrowserSpeechVoiceDropdown } from './useBrowserSpeechVoiceDropdown';
+import { useLanguageCodeForFilter } from './store-module-browser';
+
+// languages are defined as a JSON file
+import languages from './preSelect/Languages.json';
+
+export function BrowserSpeechSettings() {
+  // state
+  const [testUtterance, setTestUtterance] = React.useState<string | null>(null);
+  const [voiceNameFilters, setVoiceNameFilters] = React.useState<string[] | null>(null);
+
+  // external state
+  const [languageCode, setLanguageCode] = useLanguageCodeForFilter();
+
+  React.useEffect(() => {
+    if (languageCode) {
+      const fetchFunction = async () => {
+        let res = await fetch(`https://raw.githubusercontent.com/HadrienGardeur/web-speech-recommended-voices/refs/heads/main/json/${languageCode}.json`);
+        let data = await res.json();
+        let voices = data.voices;
+        voices = voices.filter((voice: any) => {
+          return voice.quality.includes('high') || voice.quality.includes('veryHigh');
+        });
+        let voiceNameFilters = voices.map((voice: any) => voice.name);
+        setTestUtterance(data.testUtterance);
+        setVoiceNameFilters(voiceNameFilters);
+      };
+      fetchFunction().catch((err) => {
+        console.log('Error getting voice list: ', err);
+        addSnackbar({ key: 'browser-speech-synthesis', message: 'Error getting voice list', type: 'issue' });
```

Comment:
I got this message with some of the languages of the list. Strange because I thought the list will have all valid languages.

Reply:
Upstream error, the language listed in json file does not have a corresponding file. so 404. I'll delete all invalid languages